nixos/modules/services/logging/logrotate.nix

```diff
@@ -5,26 +5,71 @@ with lib;
 let
   cfg = config.services.logrotate;
 
-  configFile = pkgs.writeText "logrotate.conf"
-    cfg.config;
+  pathOptions = {
+    options = {
+      user = mkOption {
+        type = types.str;
+        description = "The user account to use for rotation";
+      };
+      group = mkOption {
+        type = types.str;
+        description = "The group to use for rotation";
+      };
+      frequency = mkOption {
+        type = types.enum [
+          "daily" "weekly" "monthly"
+        ];
+        default = "daily";
+        description = "How often to rotate the logs";
+      };
+      keep = mkOption {
+        type = types.int;
+        default = 20;
+        description = "How many rotations to keep";
+      };
+    };
+  };
+
+  pathConfig = path: options: ''
+    "${path}" {
+      su ${options.user} ${options.group}
+      ${options.frequency}
+      missingok
+      notifempty
+      rotate ${toString options.keep}
+    }
+  '';
+
+  configFile = pkgs.writeText "logrotate.conf" (
+    (concatStringsSep "\n" (mapAttrsToList pathConfig cfg.paths)) +
+    cfg.config
+  );
 
 in
 {
   options = {
     services.logrotate = {
-      enable = mkOption {
-        type = lib.types.bool;
-        default = false;
-        description = ''
-          Enable the logrotate cron job
-        '';
+      enable = mkEnableOption "Enable the logrotate cron job";
```

Comment:
Not actually a `cron` job (anymore). So you mind fixing while you're in here?

Reply:
I didn't realize they moved away from `cron`.  Can you provide a link with the new documentation?

Reply:
Sure, just a `systemd` `timer` `unit`: https://www.freedesktop.org/software/systemd/man/systemd.timer.html
I'd simply change the description to `Whether to enable the logrotate systemd service and timer`, or something along those lines.

btw `mkEnableOption` is defined as:
```
  mkEnableOption =
    # Name for the created option
    name: mkOption {
    default = false;
    example = true;
    description = "Whether to enable ${name}.";
    type = lib.types.bool;
  };
```
Hence the `"Enable` text is redundant.